### PR TITLE
My js workflow

### PR DESCRIPTION
**Pull request by dmerejkowsky**

### CODE REVIEW

File: content/post/2017-10-18-my-javascript-workflow.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+tags: [javascript, neovim]
+---
+
+Following the (relative) success of [How I lint My Python]({{< relref "post/2017-04-14-how-i-lint-my-python.md" >}}), today I'd like to talk about the tools and technique I use while writing Javascript at work.
+
+<!-- more -->
+
+# The cycle
+
+Regular readers of this blog wont' be surprised by the fact I'm using <abbr title="Test Driven Development">TDD</abbr> and thus that I already have a "red", "green", "refactor" cycle.
```
Comment (blastrock) [inline content/post/2017-10-18-my-javascript-workflow.md]: won't*

File: content/post/2017-10-18-my-javascript-workflow.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    expect(clear).to.eq(message)
+
+    expectFailWith(bobPhone.decrypt(encrypted), /Device is revoked/);
+  });
+
+});
+
+```
+
+The whole test suite takes a few minutes to run (we have a pretty big suite of integration tests)
```
Comment (blastrock) [inline content/post/2017-10-18-my-javascript-workflow.md]: final period

File: content/post/2017-10-18-my-javascript-workflow.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+```
+
+```console
+$ git ro
+...
+```
+
+# Creating the merge request
+
+Finally it's time to create a merge request. For this I use [tsrc]() which is the tool we use to help us manage several git repositories and contains some nice features levaring the GitLab API:
```
Comment (blastrock) [inline content/post/2017-10-18-my-javascript-workflow.md]: leveraging?

File: content/post/2017-10-18-my-javascript-workflow.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+=> Running git push
+...
+=> Creating merge request
+=> Assigning to Théo
+:: See merge request at http://gitlab.dev/Tanker/SDK/merge_requests/431
+```
+
+# Accepting the merge request
+
+Our GitLab configuration does not allow anyone to push directly to master, and prevent us from merging if the CI does not pass.
```
Comment (blastrock) [inline content/post/2017-10-18-my-javascript-workflow.md]: prevents*

File: content/post/2017-10-18-my-javascript-workflow.md [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+* *refactor*: consider the mess you've made, and clean it up.
+
+My cycle when developing Javascript encapsulates this workflow:
+
+
+1. Select a feature or a bug fix to implement
+2. Write tests or production code
+3. Run the tests
+4. Back to step 2 until I'm done with the feature or bug
+5. Add flow annotations
```
Comment (cgestes) [inline content/post/2017-10-18-my-javascript-workflow.md]: Why would you flow later?

File: content/post/2017-10-18-my-javascript-workflow.md [hunk 6]
```diff
@@ -6,40 +6,41 @@ title: My Javascript Workflow
...
 8. Go back to step 6 if required
 9. Tell GitLab to merge the changes when the CI passes
 
-I'm going to go through this steps one by one, and you'll see how the tools I use are tailored for the specific
-case.
+{{< note >}}
+I'm adding `flow` annotations *after* the whole TDD cycle. This is probably because I'm used to work with dynamically typed languages, so I'm still not used to static typing. Thus, I deal with types in a separate phase. If you come to "flowed" Javascript from a C++ background, you may prefer adding types first. I've also found that, when you don't have any tests, `flow` can be of great help during refactoring.
+{{</ note >}}
 
+Anyway, let's go through this steps one by one. You will see how the tools I use are tailored for the specific task.
```
Comment (blastrock) [inline content/post/2017-10-18-my-javascript-workflow.md]: these*

for each* specific task?

**Pull request merged**